src/Controller/Api/PetController.php

```diff
@@ -50,60 +40,52 @@ public function show(Pet $pet)
         $jsonContent = $serializer->serialize($pet, 'json', [
-            'circular_reference_handler' => function ($object) {
-                return $object->getId();
-            }
+            'ignored_attributes' => ['owner' => 'address']
         ]);
-
+        
         return new Response($jsonContent);
     }
 
     public function new(Request $request)
     {
-        $owner = $this->getDoctrine()->getRepository('App\Entity\User')->find($request->get('owner'));
+        // reference: https://symfonycasts.com/screencast/symfony-rest/form-post
+        $data = json_decode($request->getContent(), true);
         
         $pet = new Pet();
-        $pet->setName($request->get('name'));
-        $pet->setDateBirth($request->get('dateBirth'));
-        $pet->setWeigth($request->get('weigth'));
-        $pet->setType($request->get('type'));
-        $pet->setBreed($request->get('breed'));
-        $pet->setOwner($owner);
-        
+        $form = $this->createForm(PetType::class, $pet);
+        $form->submit($data);
+
         $entityManager = $this->getDoctrine()->getManager();
-        $entityManager->persist($pet);
         $entityManager->flush();
+
+        $response = new Response('Pet created whit success!', Response::HTTP_CREATED);
+        $response->headers->set('Location', '/some/programmer/url');
         
-        return new JsonResponse(['msg' => 'Pet created whit success!'], Response::HTTP_OK);
+        return $response;
     }
     
-    public function edit(Request $request, $pet)
+    public function edit(Request $request, Pet $pet)
     {
-        $owner = $this->getDoctrine()->getRepository('App\Entity\User')->find($request->get('owner'));
-        
-        if (empty($pet)) {
-            return new JsonResponse(['msg' => 'Pet not found!'], Response::HTTP_NOT_FOUND);
-        }
+        // reference: https://symfonycasts.com/screencast/symfony-rest/form-post
+        $data = json_decode($request->getContent(), true);
         
-        $pet->setName($request->get('name'));
-        $pet->setDateBirth($request->get('dateBirth'));
-        $pet->setWeigth($request->get('weigth'));
-        $pet->setType($request->get('type'));
-        $pet->setBreed($request->get('breed'));
-        $pet->setOwner($owner);
-
-        if (!empty($pet->getPetName())) {
-            $entityManager = $this->getDoctrine()->getManager();
-            $entityManager->flush();
+        $pet = new Pet();
+        $form = $this->createForm(PetType::class, $pet);
+        $form->submit($data);
 
-            return new JsonResponse(['msg' => 'Pet edited whit success!'], Response::HTTP_OK);
-        }
+        $entityManager = $this->getDoctrine()->getManager();
+        $entityManager->flush();
 
-        return new JsonResponse(['msg' => 'Check the empty fields'], Response::HTTP_NOT_ACCEPTABLE);
+        $response = new Response('Pet edited whit success!', Response::HTTP_OK);
+        $response->headers->set('Location', '/some/programmer/url');
```

Comment:
`/some/programmer/url`?

Reply:
On line 80: In a video lesson, is done a recommendation for send an URL where consumer can view the entered data, but i forgot of alter. Should i keep, fix or remove?

Reply:
For now, could you please remove?